cli/src/config.rs

```diff
@@ -170,27 +170,27 @@ pub fn read_configuration(path: &Path) -> Result<Config> {
     Ok(config)
 }
 
-pub fn find_project_conf(starting_directory: &Path) -> Option<PathBuf> {
-    let mut path: PathBuf = starting_directory.into();
-    let mut attempts = 0;
-    const MAX_DEPTH: u8 = 32;
-
-    loop {
-        let search_path = path.join(PROJ_CONF_FILE);
-        if search_path.is_file() {
-            return Some(search_path);
+pub fn find_project_conf(
+    starting_directory: impl AsRef<Path>,
+    recurse_upwards: bool,
+) -> Option<PathBuf> {
+    let max_depth = if recurse_upwards { 32 } else { 1 };
+    let mut path = starting_directory.as_ref();
+
+    for _ in 0..max_depth {
```

Comment:
I've decided to just rewrite the loop here completely. It's shorter and using `parent` should be easier to understand and at the same time aborts early instead of pointlessly looping when we're already at the root.